### PR TITLE
chore: add workspaceProxy to helm chart

### PR DESCRIPTION
**Pull request by deansheather**

### CODE REVIEW

File: helm/tests/testdata/workspace_proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+coder:
+  workspaceProxy: true
+
+  image:
+    tag: latest
+
+  env:
+    - name: CODER_PRIMARY_ACCESS_URL
+      value: "https://dev.coder.com"
```
Comment (Emyrk) [inline helm/tests/testdata/workspace_proxy.yaml:9]: Should this be hardcoded?

**Review by Emyrk — APPROVED** [review]
Looks legit, just that one question

File: helm/tests/testdata/workspace_proxy.yaml (hunk 1 above)
Comment (deansheather) [reply helm/tests/testdata/workspace_proxy.yaml:9]: This is a test file, not the actual values.yaml file.

File: helm/values.yaml [hunk 2]
```diff
@@ -1,10 +1,27 @@
...
+  #
+  # Make sure you set CODER_PRIMARY_ACCESS_URL and CODER_PROXY_SESSION_TOKEN in
+  # the environment below. You can get a proxy token using the CLI:
+  #   coder proxy create \
+  #     --name "proxy-name" \
+  #     --display-name "Proxy Name" \
+  #     --icon "/emojis/xyz.png"
+  #
+  # This is an Enterprise feature. Contact sales@coder.com
+  workspaceProxy: false
```
Comment (bpmct) [inline helm/values.yaml:23]:
```suggestion
  # coder.workspaceProxy -- Whether or not this deployment of Coder is a Coder
  # Workspace Proxy. Workspace Proxies reduce the latency between the user and
  # their workspace for web connections (workspace apps and web terminal) and
  # proxied connections from the CLI. Workspace Proxies are optional and only
  # recommended for geographically sparse teams.
  #
  # Make sure you set CODER_PRIMARY_ACCESS_URL and CODER_PROXY_SESSION_TOKEN in
  # the environment below. You can get a proxy token using the CLI:
  #   coder proxy create \
  #     --name "proxy-name" \
  #     --display-name "Proxy Name" \
  #     --icon "/emojis/xyz.png"
  #
  # This is an Enterprise feature. Contact sales@coder.com
  # Docs: https://coder.com/docs/v2/latest/admin/workspace-proxies
```

**Review by bpmct — APPROVED** [review]

**Pull request merged**